libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.java

```diff
@@ -398,4 +401,30 @@ public void onReceive(Context context, Intent intent) {
             }
         }
     }
+
+    /**
+     * Callbacks for SP authentication flow.
+     *
+     * @author bhariharan
+     */
+    public class SPAuthCallback {
+
+        /**
+         * Called when the flow was successful and token response is received.
+         *
+         * @param tokenResponse Token response.
+         */
+        public void receivedTokenResponse(OAuth2.TokenEndpointResponse tokenResponse) {
+            webviewHelper.onAuthFlowComplete(tokenResponse);
+        }
+
+        /**
+         * Called when the flow was not successful.
+         *
+         * @param errorMessage Error message.
+         */
+        public void receivedErrorResponse(String errorMessage) {
+            Toast.makeText(LoginActivity.this, errorMessage, Toast.LENGTH_LONG).show();
```

Comment:
Show a `Toast` for errors -> same handling as other errors with login like SSL errors today.